### PR TITLE
chore(ci): enable partial Ivy compilation for publishable packages

### LINKED ISSUE #742: Enable Ivy partial compilation
# Problem Solved By The Feature

Ngcc will skip RxAngular packages.

# Additional Context

It's a breaking change since it targets Angular v12 and Ivy only, more info [here](https://blog.ninja-squad.com/2021/01/27/angular-linker/)

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

**Comment by edbzn** [comment]
Nice! I think it should be described as a breaking change in the commit message since users will have to use Angular >= 12 and Ivy.

**Review by edbzn — COMMENTED** [review]
Nice! I think it should be described as a breaking change in the commit message since users will have to use Angular >= 12 + Ivy.

**Comment by arturovt** [comment]
> Nice! I think it should be described as a breaking change in the commit message since users will have to use Angular >= 12 + Ivy.

Merci pour review, I'll squash commits and amend the message soon!

**Comment by BioPhoton** [comment]
I'm not sure if we should go ng12 + ivy only.
will have a think

**Comment by BioPhoton** [comment]
I would say we wait for this one a little bit. There are quite some projects using rxa ATM that will not go v 12 soon.

Especially some of the low-end device projects.

**Comment by hoebbelsB** [comment]
@arturovt @BioPhoton 
we just had a discussion about this topic in a meet. This won't be a problem for those people, just saving build times as far as I understood. correct me if i'm wrong pls

**Comment by edbzn** [comment]
Using partial compilation saves compile time but relies on the Angular Linker introduced in Angular CLI version 11.1, not version 12 as I mentioned before. It's still a breaking change but should be introduced IMO.

**Comment by wSedlacek** [comment]
@arturovt @BioPhoton @hoebbelsB
Angular v10 has LTS support ending on Dec 24, 2021.
Source: https://angular.io/guide/releases#support-policy-and-schedule

At that point would it make sense to introduce this change?

File: libs/cdk/tsconfig.lib.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "lib": ["dom", "es2018"]
   },
   "angularCompilerOptions": {
-    "enableIvy": false,
+    "enableIvy": true,
```
Comment (wSedlacek) [inline libs/cdk/tsconfig.lib.json:11]: `enableIvy` appears to have been removed as a compiler option on the latest versions of Angular.

https://github.com/angular/angular/commit/4ed8dc858aebfdf0ab271ec8c710e60f3d10058a#diff-0a65f1d37c6a41d9b8c6e91e484e9e58bb815f22e9b3425432c086c1b904eefa


The nx migration screen to 13.2 automatically removes this option too.

Comment (edbzn) [reply libs/cdk/tsconfig.lib.json:11]: We also have to remove umd options like `umdModuleIds`.

**Comment by edbzn** [comment]
Just noticed @hoebbelsB @BioPhoton, in master, we migrated to ng-packagr 13, this version does not build to ViewEngine anymore and compiles using the partial mode by default. I would suggest cleaning this PR asap instead of reverting to ng-packagr 12, what do you think?

File: libs/cdk/tsconfig.lib.json (hunk 1 above)
Comment (edbzn) [reply libs/cdk/tsconfig.lib.json:11]: Nevermind, it's already done :+1:

**Comment by hoebbelsB** [comment]
close because obsolete with v13. we just need to release one more time with enableIvy: false

**Pull request closed without merging**